src/_guides/libraries/testing.md

```diff
@@ -0,0 +1,130 @@
+---
+layout: default
+title: "Dart Testing"
+description: "How to test Dart, Flutter, and 'Dart for the Web' applications?"
+---
+
+Software testing, an important part of app development, helps verify that
+your app is working correctly before unleashing it on your users.
+This Dart testing guide outlines several types of testing, and points
+you to where you can learn more about testing on the Flutter, Dart VM,
+and 'Dart for the Web' platforms.
+
+## Testing types
+
+The Dart testing docs focus on three types of testing, out of the
+[many testing types](http://www.aptest.com/testtypes.html) that
+you may be familiar with: unit, widget (or component, depending on the
+platform), and integration (or end-to-end). Testing terminology varies,
+but these are the terms and concepts that you are likely to
```

Comment:
I'd prefer linking to https://en.wikipedia.org/wiki/Software_testing, since it clearly distinguishes between testing **levels** and **testing** types. We're trying to comment here _mainly_ about testing levels, so to avoid confusion with the Wikipedia page, I'd use the word "kind", as in:

---
The Dart testing docs focus on three kinds of testing, out of the [many testing kinds](https://en.wikipedia.org/wiki/Software_testing) that you may be familiar with: unit, component / widget, and end-to-end (a form of integration testing). Testing terminology varies, but ...

---